pkgs/_test_macros/lib/declare_x_macro.dart

```diff
@@ -12,7 +12,7 @@ class DeclareX {
 // TODO(davidmorgan): this is a placeholder; make it do something, test it.
 class DeclareXImplementation implements Macro {
   @override
-  MacroDescription get description => MacroDescription(runsInPhases: [3]);
+  MacroDescription get description => MacroDescription(runsInPhases: [2]);
```

Comment:
It seems wrong for users to be creating this description, with explicit phase numbers like this

Reply:
Yes, added a TODO on the `Macro` interface that `MacroDescription` should be populated (as far as possible) by `macro_builder` which can figure out the information and inject it in the bootstrap script, rather than asking the user macro code to do it.